src/atomic_ptr.hpp

```diff
@@ -189,13 +198,20 @@ namespace zmq
 
     private:
 
+#if defined ZMQ_ATOMIC_CXX11
+        std::atomic<T*> ptr;
+#else
         volatile T *ptr;
+#endif
+
 #if defined ZMQ_ATOMIC_PTR_MUTEX
         mutex_t sync;
 #endif
 
+#if ! defined ZMQ_ATOMIC_CXX11
         atomic_ptr_t (const atomic_ptr_t&);
```

Comment:
Note that the presence of a non-static `std::atomic` data member implicitly defines these special member functions as deleted.